src/lib.rs

```diff
@@ -983,44 +1047,25 @@ fn parse_headers_iter_uninit<'a, 'b>(
                     count += bytes.pos();
                     bytes.slice();
                     continue 'whitespace_after_colon;
-                } else {
-                    if !is_header_value_token(b) {
-                        if b == b'\r' {
-                            expect!(bytes.next() == b'\n' => Err(Error::HeaderValue));
-                        } else if b != b'\n' {
-                            return Err(Error::HeaderValue);
-                        }
+                } else if is_header_value_token(b) {
+                    break 'whitespace_after_colon;
+                }
 
-                        if config.allow_obsolete_multiline_headers_in_responses {
-                            match bytes.peek() {
-                                None => {
-                                    // Next byte may be a space, in which case that header
-                                    // is using obsolete line folding, so we may have more
-                                    // whitespace to skip after colon.
-                                    return Ok(Status::Partial);
-                                }
-                                Some(b' ') | Some(b'\t') => {
-                                    // The space will be consumed next iteration.
-                                    continue 'whitespace_after_colon;
-                                }
-                                _ => {
-                                    // There is another byte after the end of the line,
-                                    // but it's not whitespace, so it's probably another
-                                    // header or the final line return. This header is thus
-                                    // empty.
-                                },
-                            }
-                        }
+                if b == b'\r' {
+                    b = next!(bytes);
+                }
+                if b != b'\n' {
+                    continue_with_new_header_on_next_line!(bytes, b, HeaderValue);
+                }
```

Comment:
I think this changes the behavior so that a bare `\r` would be treated as a valid line ending while in the `whitespace_after_colon` loop, e.g., `"HTTP/1.1 200 OK\r\nFoo: \r\r\n"`

Reply:
Fixed.